data-products/src/main/scala/org/sunbird/analytics/model/report/TextbookProgressModel.scala

```diff
@@ -0,0 +1,195 @@
+                               creator: String, createdOn: String, slug: String)
+
+case class FinalOutput(identifier: String, aggregatedReport: Option[AggregatedReport], liveReport: Option[LiveReport], nonLiveStatusReport: Option[NonLiveStatusReport]) extends AlgoOutput with Output
+
+object TextbookProgressModel extends IBatchModelTemplate[Empty , TenantInfo, FinalOutput, FinalOutput] with Serializable {
+
+  implicit val className: String = "org.ekstep.analytics.model.TextbookProgressModel"
+  override def name: String = "TextbookProgressModel"
+
+  override def preProcess(events: RDD[Empty], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[TenantInfo] = {
+    CommonUtil.setStorageConf(config.getOrElse("store", "local").toString, config.get("accountKey").asInstanceOf[Option[String]], config.get("accountSecret").asInstanceOf[Option[String]])
+    val readConsistencyLevel: String = AppConf.getConfig("assessment.metrics.cassandra.input.consistency")
+
+    val sparkConf = sc.getConf
+      .set("spark.cassandra.input.consistency.level", readConsistencyLevel)
+      .set("spark.sql.caseSensitive", AppConf.getConfig(key = "spark.sql.caseSensitive"))
+    implicit val spark: SparkSession = SparkSession.builder.config(sparkConf).getOrCreate()
+
+    val tenantEncoder = Encoders.product[TenantInfo]
+    val data = CourseUtils.getTenantInfo(spark, loadData)
+    val id = data.as[TenantInfo](tenantEncoder).rdd
+    id
+
+  }
+
+  override def algorithm(data: RDD[TenantInfo], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[FinalOutput] = {
+    val result = data.collect().map{f =>
+     val finalResult = getContentData(f.id, f.slug)
+      finalResult
+    }.reduce((a,b) => a)
+    result
+  }
+
+  override def postProcess(data: RDD[FinalOutput], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[FinalOutput] = {
+    implicit val sqlContext = new SQLContext(sc)
+    import sqlContext.implicits._
+
+    if(data.count() > 0) {
+      val configMap = config("reportConfig").asInstanceOf[Map[String, AnyRef]]
+      val reportConfig = JSONUtils.deserialize[ReportConfig](JSONUtils.serialize(configMap))
+
+      val aggregated = data.map{f => f.aggregatedReport.getOrElse(AggregatedReport("","",Seq(),"","",0,0,0,0,0,0,0,0,0,"","Unknown"))}
+      val liveStatus = data.map(f => f.liveReport.getOrElse(LiveReport("","",List(""),"","","",0,"","","Unknown")))
+      val nonLiveStatus = data.map(f => f.nonLiveStatusReport.getOrElse(NonLiveStatusReport("","",List(""),"","","","","","","Unknown")))
+      reportConfig.output.map{f =>
+        val aggDF = aggregated.toDF()
+        CourseUtils.postDataToBlob(aggDF, f, config)
+
+        val liveStatusDF = liveStatus.toDF()
+        CourseUtils.postDataToBlob(liveStatusDF, f, config)
+
+        val nonLiveStatusDF = nonLiveStatus.toDF()
+        CourseUtils.postDataToBlob(nonLiveStatusDF, f, config)
```

Comment:
Can we also get some stats on records count per channel/slug, time taken per channel etc here which would help in debugging?

Reply:
Done